src/main/java/com/google/sps/dao/AssignmentDao.java

```diff
@@ -89,4 +94,112 @@ public ValidationResponse createAssignment(Assignment assignment) {
         }
         return assignments;
     }
+
+    @Override
+    @SneakyThrows
+    public ValidationResponse updateAssignmentForStudent(Assignment newAssignment) {
+
+        ValidationResponse.Builder validationResponseBuilder = null;
+
+        Assignment oldAssignment = getAssignmentById(newAssignment.getId());
+        newAssignment.setId(oldAssignment.getId());
```

Comment:
We don't have to do this as id should already be there in newAssignment.

Reply:
done